zerver/views/messages.py

```diff
@@ -369,7 +370,19 @@ def highlight_string_bytes_offsets(text, locs):
         result += highlight_stop
         pos = offset + length
     result += string[pos:]
-    return force_text(result)
+    result_string = force_text(result)
+    result_string = escape_data_user_email(result_string)
+    return result_string
+
+def escape_data_user_email(text):
+    # type: (AnyStr) -> Text
+    user_email_group = re.search('<span class="user-mention" data-user-email="<span class="highlight">(.*?)</span>', text)
+    if user_email_group:
+        user_email = user_email_group.group(1)
+        incorrect_string = '<span class="user-mention" data-user-email="<span class="highlight">' + str(user_email) + '</span>'
+        correct_string = '<span class="user-mention" data-user-email="' + str(user_email)
+        text = text.replace(incorrect_string, correct_string)
+    return text
```

Comment:
This will probably fix this case, but I think we want to work on a solution that understands the general case of this sort of HTML tag, rather than just this specific case.  E.g. I bet we have a similar bug with emoji.